free_fleet/src/requests/RequestInfo.hpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef SRC__REQUESTS__REQUESTINFO_HPP
+#define SRC__REQUESTS__REQUESTINFO_HPP
+
+#include <utility>
+#include <rmf_traffic/Time.hpp>
+#include <rmf_utils/optional.hpp>
+#include <free_fleet/agv/RobotInfo.hpp>
+
+namespace free_fleet {
+namespace requests {
+
+//==============================================================================
+class RequestInfo
+{
+public:
+
+  /// Time stamp of when the request was initialized.
+  virtual rmf_traffic::Time init_time() const = 0;
+
+  /// Whether the request has been acknowledged by the robot.
+  virtual bool acknowledged() const = 0;
+
+  /// Time stamp of when the request was acknowledged.
+  ///
+  /// \return
+  ///   Returns a nullopt if the request has not been acknowledged.
+  virtual rmf_utils::optional<rmf_traffic::Time> acknowledged_time() const = 0;
+
+  /// Sets the time that this request was acknowledged.
+  virtual void acknowledged_time(rmf_traffic::Time time) = 0;
```

Comment:
Thanks for mentioning this! I read up a bit on best practices for abstract interface writing, I've since modified the derived class to accept a `std::function<rmf_traffic::Time()>` in the constructor instead, while the abstract interface just has a `acknowledge_request()` function to be implemented. Let me know what you think about this approach.

* https://github.com/open-rmf/free_fleet/pull/76/commits/aa011536f1a53cd8c3bf0ff0becbc02a1be9bcda#diff-658354c6c98453006395b6f94f5e1bbd9e25e6b678aaddd0f6d0930b1244d14cR50-R51
* https://github.com/open-rmf/free_fleet/pull/76/commits/aa011536f1a53cd8c3bf0ff0becbc02a1be9bcda#diff-4c02fa97a8b4a27767d73f47923becde76b80bbbffed1efdd36bf95247693978R36-R48